test/test_misc.py

```diff
@@ -2520,6 +2520,74 @@ def fn():
         res = opt_fn()
         self.assertTrue(same(ref, res))
 
+    def test_autograd_function_equivalence(self):
```

Comment:
Also check the result of these calls just to make sure it's working as we expect.